server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java

```diff
@@ -0,0 +1,300 @@
+
+    context.prepareUpsert(deleteRequest(toBeDeleted)).execute().commit();
+  }
+
+  private List<Long> migrateProjectsWebhooks(Context context, Map<String, PropertyRow> properties, Map<Long, String> projectsKeys) throws SQLException {
+    List<Long> toBeDeleted = new ArrayList<>();
+    PropertyRow index = properties.get("sonar.webhooks.project");
+    if (index == null) {
+      return emptyList();
+    }
+
+    // can't lambda due to checked exception.
+    for (Webhook webhook : extractProjectWebhooksFrom(properties, index.value().split(","), projectsKeys)) {
+      toBeDeleted.addAll(insert(context, webhook));
+    }
+
+    toBeDeleted.add(index.id());
+    return toBeDeleted;
+  }
+
+  private List<Long> migrateGlobalWebhooks(Context context, Map<String, PropertyRow> properties) throws SQLException {
+
+    List<Long> toBeDeleted = new ArrayList<>();
+    PropertyRow index = properties.get("sonar.webhooks.global");
+    if (index == null) {
+      return emptyList();
+    }
+
+    // can't lambda due to checked exception.
+    for (Webhook webhook : extractGlobalWebhooksFrom(context, properties, index.value().split(","))) {
+      toBeDeleted.addAll(insert(context, webhook));
+    }
+
+    toBeDeleted.add(index.id());
+    return toBeDeleted;
+  }
+
+  private List<Long> insert(Context context, Webhook webhook) throws SQLException {
+
+    if (webhook.isValid()) {
+      context.prepareUpsert("insert into webhooks (uuid, name, url, organization_uuid, project_uuid, created_at, updated_at) values (?, ?, ?, ?, ?, ?, ?)")
+        .setString(1, uuidFactory.create())
+        .setString(2, webhook.name())
+        .setString(3, webhook.url())
+        .setString(4, webhook.organisationUuid())
+        .setString(5, webhook.projectUuid())
+        .setLong(6, webhook.createdAt())
+        .setLong(7, webhook.createdAt())
+        .execute()
+        .commit();
+    } else {
+      LOGGER.info("Unable to migrate inconsistent webhook (entry deleted from PROPERTIES) : " + webhook);
+    }
+
+    return webhook.rowIds();
+  }
+
+  private static String deleteRequest(List<Long> toBeDeleted) {
+    return new StringBuilder()
+      .append("DELETE from properties WHERE id in (")
+      .append(Joiner.on(',').join(toBeDeleted))
+      .append(")")
+      .toString();
+  }
+
+  private List<Webhook> extractGlobalWebhooksFrom(Context context, Map<String, PropertyRow> properties, String[] values) throws SQLException {
+    String defaultOrganizationUuid = defaultOrganizationUuidProvider.get(context);
+    return Arrays.stream(values)
+      .map(value -> new Webhook(
+        properties.get("sonar.webhooks.global." + value + ".name"),
+        properties.get("sonar.webhooks.global." + value + ".url"),
+        defaultOrganizationUuid, null))
+      .collect(toList());
+  }
+
+  private static List<Webhook> extractProjectWebhooksFrom(Map<String, PropertyRow> properties, String[] values, Map<Long, String> projectsKeys) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "extractProjectWebhooksFrom" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWGP8T0uFMFpNJFuf3Az&open=AWGP8T0uFMFpNJFuf3Az)

[ISSUE_KEY]: AWGP8T0uFMFpNJFuf3Az